solarforecastarbiter/datamodel.py

```diff
@@ -0,0 +1,302 @@
+    SingleAxisModelingParameters
+    """
+    ac_power: float
+    dc_power: float
+    temperature_coefficient: float
+
+
+@dataclass(frozen=True)
+class FixedTiltModelingParameters(PVModelingParameters):
+    """
+    A class based on PVModelingParameters that has additional parameters
+    for fixed tilt PV systems.
+
+    Parameters
+    ----------
+    surface_tilt : float
+        Tilt from horizontal of a fixed tilt system, degrees
+    surface_azimuth : float
+        Azimuth angle of a fixed tilt system, degrees East of North
+
+
+    See Also
+    --------
+    PVModelingParameters
+    """
+    surface_tilt: float
+    surface_azimuth: float
+
+
+@dataclass(frozen=True)
+class SingleAxisModelingParameters(PVModelingParameters):
+    """
+    A class based on PVModelingParameters that has additional parameters
+    for single axis tracking systems.
+
+    Parameters
+    ----------
+    axis_tilt : float
+        Tilt from horizontal of the tracker axis, degrees
+    axis_azimuth : float
+        Azimuth angle of the tracker axis, degrees East of North
+    ground_coverage_ratio : float
+        Ratio of total width of modules on a tracker to the distance between
+        tracker axes. For example, for trackers each with two modules of 1m
+        width each, and a spacing between tracker axes of 7m, the ground
+        coverage ratio is 0.286(=2/7).
+    backtrack : bool
+        Indicator of if a tracking system uses backtracking
+    maximum_rotation_angle : float
+        maximum rotation from horizontal of a single axis tracker, degrees
+
+    See Also
+    --------
+    PVModelingParameters
+    """
+    axis_tilt: float
+    axis_azimuth: float
+    ground_coverage_ratio: float
+    backtrack: bool
+    maximum_rotation_angle: float
+
+
+@dataclass(frozen=True)
+class SingleAxisPowerPlant(Site):
+    """
+    Class for keeping track of metadata associated with single axis tracking
+    solar power plant Sites. Adds additional parameters to the Site dataclass.
+
+    Parameters
+    ----------
+    modeling_parameters : SingleAxisModelingParameters
+        Modeling parameters for a single axis system
+
+    See Also
+    --------
+    Site
+    SingleAxisModelingParameters
+    """
+    modeling_parameters: SingleAxisModelingParameters = field(
+        default_factory=SingleAxisModelingParameters)
+
+
+@dataclass(frozen=True)
+class FixedTiltPowerPlant(Site):
+    """
+    Class for keeping track of metadata associated with fixed tilt tracking
+    solar power plant Sites. Adds additional parameters to the Site dataclass.
+
+    Parameters
+    ----------
+    modeling_parameters : FixedTiltModelingParameters
+        Modeling parameters for a fixed tilt system
+
+    See Also
+    --------
+    Site
+    FixedTiltModelingParameters
+    """
+    modeling_parameters: FixedTiltModelingParameters = field(
+        default_factory=FixedTiltModelingParameters)
+
+
+@dataclass(frozen=True)
+class UnitsSetter(GetItem):
```

Comment:
A `UnitsSetter` class is a weird thing to inherit. I think it's relatively likely that we'd regret this one. Aren't we supposed to prefer functions to classes? What if I want `UnitsSetter` but not `GetItem`?

Reply:
I don't think we'll regret automatically setting the units from the variable type. now handled by a function

Reply:
and we don't want to modify the dataclass after creation, so should limit any adjustments to __post_init__ and not pass the Forecast object to the function to assign the units

Reply:
> I don't think we'll regret automatically setting the units from the variable type.

I agree, my comment was about the inheritance.

> now handled by a function

I like this better.

> and we don't want to modify the dataclass after creation, so should limit any adjustments to **post_init** and not pass the Forecast object to the function to assign the units

makes sense